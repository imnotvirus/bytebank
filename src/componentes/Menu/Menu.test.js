import { render } from '@testing-library/react';
import Menu from './index';
import { BrowserRouter } from 'react-router-dom';

const renderComponent = (props) =>
  render(
    <BrowserRouter>
      <Menu {...props} />
    </BrowserRouter>
  );

describe('Teste do componente Menu', () => {
  it('deve renderizar o link para pagina inicial', () => {
    const { getByText } = renderComponent();
    const linkElement = getByText(/início/i);
    expect(linkElement).toBeInTheDocument();
  });
  it('deve renderizar uma lista de links', () => {
    const { getAllByRole } = renderComponent();
    const links = getAllByRole('link');
    expect(links).toHaveLength(4);
  });
  it('Não deve renderizar o link para extrato', () => {
    const { queryByText } = renderComponent();
    const linkElement = queryByText(/extrato/i);
    expect(linkElement).not.toBeInTheDocument();
  });
  it('deve renderizar uma lista de links com a classe link', () => {
    const { getAllByRole } = renderComponent();
    const links = getAllByRole('link');
    links.forEach((link) => {
      expect(link).toHaveClass('link');
    });
    expect(links).toMatchSnapshot();
  });
});
