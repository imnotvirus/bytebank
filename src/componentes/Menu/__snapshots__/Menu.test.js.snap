// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste do componente Menu deve renderizar uma lista de links com a classe link 1`] = `
Array [
  <a
    class="link false"
    href="/"
  >
    Início
  </a>,
  <a
    class="link false"
    href="/cartoes"
  >
    Cartões
  </a>,
  <a
    class="link false"
    href="/servicos"
  >
    Serviços
  </a>,
  <a
    class="link false"
    href="/investimentos"
  >
    Investimentos
  </a>,
]
`;
